specifications/sep-payment-channel-mechanism.md

```diff
@@ -0,0 +1,631 @@
+guaranteed to be able to observe the blockchain state and submit transactions
+within any period of length O.
+
+The payment channel consists of a 2-of-2 multisig escrow account E, and a
+series of transaction sets that contain _declaration_ and _closing_
+transactions on E signed by both participants.  The closing transaction
+defines the final state of the channel that creates claimable balances for R
+and returns control of E to I.  Each generation of declaration and closing
+transaction sets in the series are an agreement on a new final state for the
+channel.
+
+The payment channel also uses a second 2-of-2 multisig reserve account V, to
+sponsor the claimable balances ledger entries that are created at channel
+close, and that disburse funds to R.
+
+## Specification
+
+### Participants
+
+A payment channel has two participants:
+
+- I, the _initiator_, who proposes the payment channel, performs the first
+setup step, and will be able to make deposits to the payment channel without
+coordination.  I creates escrow account E and receives disbursement through
+regaining control of E at channel close.
+
+- R, the _responder_, who joins the payment channel, and receives disbursement
+through claimable balances at channel close.
+
+### Observation Period
+
+A payment channel defines an observation period O within which all
+participants are guaranteed to be able to observe the blockchain state and
+submit transactions in response to changing state.
+
+The participants agree on the period O at channel open.
+
+The observation period is defined both as a duration in time, and a count of
+ledgers. The observation period has passed if both the duration and ledger count
+have been exceeded. These two properties together are referred to as O
+throughout the protocol.
+
+The participants may agree to change the period O at anytime by following the
+[Change the Observation Period](#Change-the-Observation-Period) process.
+
+### Accounts
+
+The payment channel utilizes two Stellar accounts that are both 2-of-2
+multisig accounts while the channel is open:
+
+- E, the _escrow account_, that holds the assets that both participants have
+contributed to the channel and that will be distributed to the participants
+at channel close according to the final close transactions submitted.
+Created by I.  Jointly controlled by I and R while the channel is open.
+Control is returned to I at close.
+
+- V, the _reserve account_, that holds an amount of native asset contributed
+by R that will be used to sponsor the claimable balance ledger entries
+created at disbursement.  Created by R.  Jointly controlled by I and R while
+the channel is open.  Control is returned to R at close.  Cannot be merged
+until all claimable balances created at close for R are claimed by R.
+
+### Constants
+
+The two participants agree on the following constants:
+
+- m, the _maximum transaction count for an iteration's transaction set_, is
```

Comment:
I think the definition of m is worthwhile because otherwise we have magic numbers, in this case 2, throughout the formulas. That was something I found difficult to understand in the original specification of the protocol in CAP-21.

---

I commented throughout about why I don't think it is suitable to use m in the other places.

> we should potentially replace `1` with `m`

Note that m has value 2.

Reply:
Hm yeah I agree that's better than magic numbers. Makes sense to me!